starter-code/scripts/views/articleView.js

```diff
@@ -18,17 +18,20 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //Populates author and category filters by appending options to their respective
+  //filters after mapping over the array of article templates and returning a new array
+  //of author names and a new array of article categories.
```

Comment:
Essentially, this method calls Article.allAuthors, which returns to us an array of author names with duplicates removed. We run a map on that array which produces an array of templates with the "val" Handlebars expression filled in with the value of the author -- the current string being iterated over. This essentially returns an array of filled-in HTML option elements that we then append to our author-filter select box. 

Then, our method calls Article.allCategories, which is a function that takes a callback function. This callback function we define inline as an anonymous function, but it doesn't get run until after Article.allCategories finishes. Article.allCategories runs a query to get unique categories from the database, then calls the callback. The callback takes in the data returned from the query, and appends to our category-filter select box the result of mapping over this array of categories and filling in templates (in a similar way to above).